cli/update.go

```diff
@@ -0,0 +1,227 @@
+package cli
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/bitrise-io/bitrise/configs"
+	"github.com/bitrise-io/bitrise/plugins"
+	"github.com/bitrise-io/bitrise/version"
+	"github.com/bitrise-io/go-utils/log"
+	ver "github.com/hashicorp/go-version"
+	"github.com/urfave/cli"
+)
+
+const (
+	tagsURL     = "https://api.github.com/repos/bitrise-io/bitrise/tags"
+	downloadURL = "https://github.com/bitrise-io/bitrise/releases/download/%s/bitrise-%s-x86_64"
+)
+
+var updateCommand = cli.Command{
+	Name:  "update",
+	Usage: "Updates the Bitrise CLI.",
+	Action: func(c *cli.Context) error {
+		log.Infof("Updating Bitrise CLI...")
```

Comment:
move the action into a separate function, for idiomatic error handling: https://github.com/bitrise-io/bitrise/blob/master/cli/workflow_list.go#L20